src/AxonIQ.AxonServer.Connector/AxonServerConnection.cs

```diff
@@ -100,25 +102,19 @@ private State CurrentState
         get => _state;
         set
         {
-            if (_state is not State.Connected && value is State.Connected)
-            {
-                OnConnected();
-            }
-
-            if (_state is not State.Disconnected && value is State.Disconnected)
-            {
-                OnDisconnected();
-            }
-
+            var connected = _state is not State.Connected && value is State.Connected;
```

Comment:
Because event handlers are triggered synchronously, they would not observe the new "Current" state (it wasn't set yet at that point).